### ISSUE TITLE
LaterStream object does not follow the Node stream convention

### ISSUE THREAD

**Issue by frankrousseau** [cozy/cozy-db#4]
It's hard to deal with the LaterStream object and associated functions. It's not a stream in a Node way, so it's disapointing to use it.

We should rename it or making it extends the native stream object.

**Comment by aenario** [comment]
Agree, it seemed like a good quickfix idea at the time but it will be cleaner to properly extends node stream.

**Comment by y-lohse** [comment]
+1, I just got bitten by this one.
